### DISCUSSION TITLE
Baf configuration

### DISCUSSION THREAD

**Discussion by ghost** [soot-oss/soot#2075] [Q&A]
i'm using Baf to create java class from Jimple . how can i preserve the original method argument names ?

**Comment by StevenArzt** [comment]
The IRs in Soot do not model parameter names because they don't exist in bytecode. In Jimple, there are tags to store extra debug information such as parameter names if they are available. Maybe these tags are kept in Baf to also write out debug information when producing new class files. Have you looked at the tags associated with the method?
